activerecord/lib/active_record/validations/uniqueness.rb

```diff
@@ -86,12 +86,21 @@ def build_relation(klass, attribute, value) # :nodoc:
       def scope_relation(record, relation)
         Array(options[:scope]).each do |scope_item|
           if reflection = record.class._reflect_on_association(scope_item)
```

Comment:
Ah, interesting. Perhaps we should use `record.association(scope_item).reader` instead of the `send` in the `if` case -- that way we'd be consistent in using the "real" value and ignoring any overrides.

What do you think?